File: backend/src/main/java/com/example/backend/service/TypeOfExaminationService.java
package com.example.backend.service;

import com.example.backend.dto.request.TypeOfExaminationRequestDTO;
import com.example.backend.model.TypeOfExamination;
import com.example.backend.model.User;

import java.util.List;

public interface TypeOfExaminationService {

    List<TypeOfExamination> getTypesOfExamination();

    TypeOfExamination createTypeOfExamination(TypeOfExaminationRequestDTO typeOfExaminationRequestDTO);

    TypeOfExamination updateTypeOfExamination(Integer id, TypeOfExaminationRequestDTO typeOfExaminationRequestDTO);

    TypeOfExamination getTypeOfExamination(Integer id);

    boolean deleteTypeOfExamination(Integer id);

    List<User> getDoctorBySpecialization(Integer id);
}

File: mdb-angular-free/src/app/service/type-of-examination.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { baseUrl, httpOptions} from './constants';
import { TypeOfExamination } from '../model/typeOfExamination';
import { Observable } from 'rxjs';
import { User } from '../model/user';

@Injectable({
  providedIn: 'root'
})
export class TypeOfExaminationService {

  constructor(private httpClient: HttpClient) { }

  getTypesOfExamination():Observable<TypeOfExamination[]> {
    return this.httpClient.get<TypeOfExamination[]>(baseUrl + 'types-of-examination', httpOptions);
  }

  createTypeOfExamination(typeOfExamination: TypeOfExamination) {
    return this.httpClient.post(baseUrl + 'types-of-examination', typeOfExamination, httpOptions);
  }

  updateTypeOfExamination(typeOfExamination: TypeOfExamination) {
    return this.httpClient.put(baseUrl + 'types-of-examination/' + typeOfExamination.id, typeOfExamination, httpOptions);
  }

  deleteTypeOfExamination(id: number) {
    return this.httpClient.delete(baseUrl + 'types-of-examination/' + id, httpOptions);
  }

  getDoctorsBySpecialiyation(id: number) {
    return this.httpClient.get<User[]>(baseUrl + 'types-of-examination/specialization/' + id, httpOptions);
  }

  getTypeOfExamination(id: number) {
    return this.httpClient.get<TypeOfExamination>(baseUrl + 'types-of-examination/' + id, httpOptions);
  }
 
}